compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -1830,7 +1830,8 @@ static FnSymbol* buildReadThisFnSymbol(AggregateType* ct, ArgSymbol** filearg, c
 
   fn->cname = astr("_auto_", ct->symbol->name, "_", name);
 
-  fn->_this = new ArgSymbol(INTENT_BLANK, "this", ct);
+  auto thisIntent = isDeserialize ? INTENT_REF : INTENT_BLANK;
+  fn->_this = new ArgSymbol(thisIntent, "this", ct);
```

Comment:
I'm also not sure what this change does wrt the deprecation?

Reply:
With ``BinaryDeserializer`` being used more, it exposed some issues with using the blank intent for the compiler-generated "deserialize" method. Add an explicit "ref" intent fixed some weird bugs.